crates/core/c8y_api/src/smartrest/inventory.rs

```diff
@@ -116,6 +117,28 @@ pub fn service_creation_message_payload(
     ]))
 }
 
+/// Create a SmartREST message to set a response interval for c8y_RequiredAvailability.
+///
+/// In the SmartREST 117 message, the interval must be in MINUTES, and can be <=0,
+/// which means the device is in maintenance mode in the c8y context.
+/// Details: https://cumulocity.com/docs/device-integration/fragment-library/#device-availability
+#[derive(Debug)]
+pub struct C8ySmartRestSetInterval117 {
+    pub c8y_topic: C8yTopic,
+    pub interval: u64,
```

Comment:
```suggestion
    pub interval: Duration,
```

Using a duration here would keep the knowledge, that c8y expects here a number of minutes, in a single place: the `impl From<C8ySmartRestSetInterval117> for MqttMessage `.

For instance, this would avoid such fragile code and comment:

```
                interval: self.config.interval.as_secs() / 60, // convert to MINUTES
```

Reply:
Done in cb41d2df39dd6dfb81f47ee784af77e08368e2f6